src/plugins/intel_npu/src/compiler/include/graph_transformations.hpp

```diff
@@ -14,9 +17,51 @@
  */
 namespace intel_npu::driverCompilerAdapter {
 
-/**
- * @brief Serialize OpenVINO model to IR
- */
-IR serializeToIR(const std::shared_ptr<const ov::Model>& model, uint32_t supportedVersionByCompiler = 7);
+class IR {
+public:
+    IR(const std::shared_ptr<const ov::Model>& origModel,
+       uint32_t supportedVersionByCompiler = 11,
+       bool largeModel = false);
+    ~IR();
+
+    std::istream& getXml() {
+        if (!_isLargeModel) {
+            return _xml;
+        } else {
+            return _xmlStream;
+        }
+    }
+
+    std::istream& getWeights() {
+        if (!_isLargeModel) {
+            return _weights;
+        } else {
+            return _weightsStream;
+        }
+    }
+
+private:
+    bool isLargeModel() const {
```

Comment:
Any reason for having a private getter here?

Reply:
Removed